server/org.eclipse.emfcloud.ecore.glsp/src/main/java/org/eclipse/emfcloud/ecore/glsp/model/EcoreModelServerAccess.java

```diff
@@ -97,16 +99,78 @@ private EPackage getEPackage(EcoreModelState modelState) {
+		return this.add(modelState, parent, EcorePackage.eINSTANCE.getEClass_EStructuralFeatures(), newEReference);
+	}
+	
+	public boolean addEAttribute(EcoreModelState modelState, EAttribute newEAttribute, EClassifier parent) {
+		return this.add(modelState, parent, EcorePackage.eINSTANCE.getEClass_EStructuralFeatures(), newEAttribute);
+	}
+	
+	public boolean addEEnumLiteral(EcoreModelState modelState, EEnumLiteral newEEnumLiteral, EEnum parent) {
+		return this.add(modelState, parent, EcorePackage.eINSTANCE.getEEnum_ELiterals(), newEEnumLiteral);
+	}
+	
+	public boolean setOpposite(EcoreModelState modelState, EReference eReference, EReference opposite) {
+		return this.set(modelState, eReference, EcorePackage.eINSTANCE.getEReference_EOpposite(), opposite);
+	}
+	
+	public boolean setInstanceName(EcoreModelState modelState, EClassifier eClassifier, String name) {
+		return this.set(modelState, eClassifier, EcorePackage.eINSTANCE.getEClassifier_InstanceClassName(), name);
+	}
+	
+	public boolean setName(EcoreModelState modelState, EClassifier eClassifier, String name) {
+		return this.set(modelState, eClassifier, EcorePackage.eINSTANCE.getENamedElement_Name(), name);
+	}
+	
+	public boolean setAttributeName(EcoreModelState modelState, EAttribute eAttribute, String name) {
+		return this.set(modelState, eAttribute, EcorePackage.eINSTANCE.getENamedElement_Name(), name);
+	}
+	
+	public boolean setAttributeType(EcoreModelState modelState, EAttribute eAttribute, EClassifier type) {
+		return this.set(modelState, eAttribute, EcorePackage.eINSTANCE.getEAttribute_EAttributeType(), type);
+	}
+	
+	public boolean setLiteralName(EcoreModelState modelState, EEnumLiteral eEnumLiteral, String literal) {
+		return this.set(modelState, eEnumLiteral, EcorePackage.eINSTANCE.getENamedElement_Name(), literal);
+	}
+	
+	public boolean setAttributeType(EcoreModelState modelState, EAttribute eAttribute, EAttribute type) {
+		return this.set(modelState, eAttribute, EcorePackage.eINSTANCE.getEAttribute_EAttributeType(), type);
+	}
+	
+	public boolean setEdgeName(EcoreModelState modelState, EReference eReference, String name) {
+		return this.set(modelState, eReference, EcorePackage.eINSTANCE.getENamedElement_Name(), name);
+	}
+	
+	public boolean setLowerMultiplicity(EcoreModelState modelState, EReference eReference, int multiplicity) {
+		return this.set(modelState, eReference, EcorePackage.eINSTANCE.getETypedElement_LowerBound(), multiplicity);
+	}
+	
+	public boolean setUpperMultiplicity(EcoreModelState modelState, EReference eReference, int multiplicity) {
+		return this.set(modelState, eReference, EcorePackage.eINSTANCE.getETypedElement_UpperBound(), multiplicity);
 	}
 
-	private boolean add(EcoreModelState modelState, EObject owner, EReference feature, EObject element) {
+	private boolean add(EcoreModelState modelState, EObject owner, EReference feature, Object addObject) {
```

Comment:
Do we actually have a use-case where we add an element that is not an EObject? (In general, it's possible when editing the value of N-ary EAttributes, but I don't think Ecore has any)

Reply:
Actually not i changed this for Strings before having the Set-Command and forgot to change it back. I reverted it.